src/database/accessors/loadProposalBundle.ts

```diff
@@ -1,16 +1,222 @@
-import { isProposal } from '../../types';
 import { loadBundle } from './loadBundle';
 import type { TinyPgParams } from 'tinypg';
 import type {
   Bundle,
   Proposal,
+  ProposalVersion,
+  ProposalFieldValue,
+  ApplicationFormField,
 } from '../../types';
 
+/* eslint-disable @typescript-eslint/no-type-alias */
+type AddPrefixToObject<T, P extends string> = {
+  [K in keyof T as K extends string ? `${P}${K}` : never]: T[K]
+};
+/* eslint-enable @typescript-eslint/no-type-alias */
+
+const filterByDistinctId = (
+  candidateItem: { id: number },
+  index: number,
+  self: { id: number }[],
+) => index === self.findIndex(
+  (item) => item.id === candidateItem.id,
+);
+
+// See https://github.com/typescript-eslint/typescript-eslint/issues/1824
+/* eslint-disable @typescript-eslint/indent */
+type ProposalPiece = AddPrefixToObject<Proposal, 'proposal_'>;
+type ProposalVersionPiece = AddPrefixToObject<ProposalVersion, 'proposalVersion_'>;
+type ProposalFieldValuePiece = AddPrefixToObject<ProposalFieldValue, 'proposalFieldValue_'>;
+type ApplicationFormFieldPiece = AddPrefixToObject<ApplicationFormField, 'applicationFormField_'>;
+/* eslint-enable @typescript-eslint/indent */
+
+const isApplicationFormFieldPiece = (
+  o: Record<string, unknown>,
+): o is ApplicationFormFieldPiece => (
+  'applicationFormField_id' in o
+);
+const isProposalFieldValuePiece = (
+  o: Record<string, unknown>,
+): o is ProposalFieldValuePiece => (
+  'proposalFieldValue_id' in o
+);
+const isProposalVersionPiece = (
+  o: Record<string, unknown>,
+): o is ProposalVersionPiece => (
+  'proposalVersion_id' in o
+);
+const isProposalPiece = (
+  o: Record<string, unknown>,
+): o is ProposalPiece => (
+  'proposal_id' in o
+);
+
+const extractApplicationFormFields = (
+  applicationFormFieldPieces: ApplicationFormFieldPiece[],
+): ApplicationFormField[] => (
+  applicationFormFieldPieces.reduce<ApplicationFormField[]>(
+    (applicationFormFields, applicationFormFieldPiece) => [
+      ...applicationFormFields,
+      {
+        id: applicationFormFieldPiece.applicationFormField_id,
+        applicationFormId: applicationFormFieldPiece.applicationFormField_applicationFormId,
+        canonicalFieldId: applicationFormFieldPiece.applicationFormField_canonicalFieldId,
+        position: applicationFormFieldPiece.applicationFormField_position,
+        label: applicationFormFieldPiece.applicationFormField_label,
+        createdAt: applicationFormFieldPiece.applicationFormField_createdAt,
+      },
+    ],
+    [],
+  ).filter(filterByDistinctId)
+);
+
+const extractProposalFieldValues = (
```

Comment:
These extractors + their "Piece" correlated types stand alone outside of proposals.  I'm not certain how it should be organized yet, but would do so before merging this PR if we go this route.